flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPool.java

```diff
@@ -659,7 +660,7 @@ public void disconnectResourceManager() {
 			.orTimeout(pendingRequest.getAllocatedSlotFuture(), allocationTimeout.toMilliseconds(), TimeUnit.MILLISECONDS)
 			.whenCompleteAsync(
 				(AllocatedSlot ignored, Throwable throwable) -> {
-					if (throwable != null) {
+					if (throwable instanceof TimeoutException) {
```

Comment:
This callback is only intended to react to timeouts of the slot allocation future. Since we return the future to the user, any other exception should be visible.